### PR TITLE
Add info about compatibility issues with Connector/J 8.0 with TiDB v7.5

### PR DESCRIPTION
**Pull request by yusra1007**
### What is changed, added or deleted? (Required)

MySQL Connector/J 8.0 has compatibility issues with TiDB v7.5. Earlier versions of TiDB are fine as they announce themselves as MySQL v5.7. Later versions are fine as they implement the information_schema.KEYWORD table.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/15794
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-choose-driver-or-orm.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ TiDB 兼容 MySQL 的协议，但存在部分与 MySQL 不兼容或有差异的
 
 按照 [MySQL 文档](https://dev.mysql.com/doc/connector-j/en/)中的说明下载并配置 Java JDBC 驱动程序即可使用。对于 TiDB v6.3.0 及以上版本，建议使用 MySQL Connector/J 8.0.33 及以上版本。
 
-> **建议：**
+> **注意：**
 >
 > 在 8.0.32 之前的 MySQL Connector/J 8.0 版本中存在一个 [bug](https://bugs.mysql.com/bug.php?id=106252)，当与 TiDB v6.3.0 之前的版本一起使用时，可能会导致线程卡死。为了避免此问题，建议使用 MySQL Connector/J 8.0.32 或更高版本，或者使用 TiDB JDBC（见 *TiDB-JDBC* 标签）。
-
+MySQL Connector/J 8.0 与 TiDB v7.5.0版本一起使用 时，建议将 TiDB 配置项 [`server-version`](https://docs.pingcap.com/tidb/v7.5/tidb-configuration-file#server-version) 设置为`"5.7.25-TiDB-v7.5.0"`。如果 TiDB 服务器报告的是 MySQL 8.0.11 或以上版本，MySQL Connector/J 会尝试访问 `information_schema.KEYWORDS` 表。但是，该表不会出现在TiDB v7.5.0 中。
```
Comment (shawn0915) [inline develop/dev-guide-choose-driver-or-orm.md:35]: 这里如果也是“建议”的一部分，前面可以加上“>”。
需要调整下空格的使用。
8.0.11-TiDB-7.5.0 意为 TiDB 7.4 开始正式兼容 MySQL 8.0 系列，如有需要，可以考虑临时变更为 8.0.10-TiDB-7.5.0，或者升级版本，预计从 TiDB 7.5.1 开始支持 I_S.KEYWORDS 表 https://github.com/pingcap/tidb/pull/48807

File: develop/dev-guide-choose-driver-or-orm.md [hunk 2]
```diff
@@ -29,10 +29,10 @@ TiDB 兼容 MySQL 的协议，但存在部分与 MySQL 不兼容或有差异的
 
 按照 [MySQL 文档](https://dev.mysql.com/doc/connector-j/en/)中的说明下载并配置 Java JDBC 驱动程序即可使用。对于 TiDB v6.3.0 及以上版本，建议使用 MySQL Connector/J 8.0.33 及以上版本。
```
Comment (shawn0915) [inline develop/dev-guide-choose-driver-or-orm.md:30]: > 8.0.33 及以上版本

是否有必要加上 <8.1 版本，毕竟 Connector/J 8.1.0 、 8.2.0 都是非 LTS 版本

Comment (Oreoxmt) [reply develop/dev-guide-choose-driver-or-orm.md:30]: 这个 PR 是翻译 https://github.com/pingcap/docs/pull/15794 的内容，所以尽量保持内容的一致性。如果有其他文档内容的修改，欢迎提新的 PR 👏

关于 MySQL Connector/J 的版本问题 @Icemap PTAL

Comment (Icemap) [reply develop/dev-guide-choose-driver-or-orm.md:30]: 这里的意见与另一意见两方相左：https://github.com/pingcap/docs/pull/16083#issuecomment-1884876450

Comment (shawn0915) [reply develop/dev-guide-choose-driver-or-orm.md:30]: @Icemap pls ignore mine, agree with [Daniël](https://github.com/dveeden), 
let's recommend to use Connector/J 8.2.0 GA release version.

File: develop/dev-guide-choose-driver-or-orm.md [hunk 3]
```diff
@@ -29,10 +29,10 @@ TiDB 兼容 MySQL 的协议，但存在部分与 MySQL 不兼容或有差异的
...
 
-> **建议：**
+> **注意：**
 >
 > 在 8.0.32 之前的 MySQL Connector/J 8.0 版本中存在一个 [bug](https://bugs.mysql.com/bug.php?id=106252)，当与 TiDB v6.3.0 之前的版本一起使用时，可能会导致线程卡死。为了避免此问题，建议使用 MySQL Connector/J 8.0.32 或更高版本，或者使用 TiDB JDBC（见 *TiDB-JDBC* 标签）。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:34]:
```suggestion
> - 在 8.0.32 之前的 MySQL Connector/J 8.0 版本中存在一个 [bug](https://bugs.mysql.com/bug.php?id=106252)，当与 TiDB v6.3.0 之前的版本一起使用时，可能会导致线程卡死。为了避免此问题，建议使用 MySQL Connector/J 8.0.32 或更高版本，或者使用 TiDB JDBC（见 *TiDB-JDBC* 标签）。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 4]
```diff
@@ -29,10 +29,10 @@ TiDB 兼容 MySQL 的协议，但存在部分与 MySQL 不兼容或有差异的
...
-
+> MySQL Connector/J 8.0 与 TiDB v7.5.0版本一起使用 时，建议将 TiDB 配置项 [`server-version`](https://docs.pingcap.com/tidb/v7.5/tidb-configuration-file#server-version) 设置为`"5.7.25-TiDB-v7.5.0"`。如果 TiDB 服务器报告的是 MySQL 8.0.11 或以上版本，MySQL Connector/J 会尝试访问 `information_schema.KEYWORDS` 表。但是，该表不会出现在TiDB v7.5.0 中。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:35]:
```suggestion
> - MySQL Connector/J 8.0 与 TiDB v7.5.0 一起使用时，建议将 TiDB 配置项 [`server-version`](/tidb-configuration-file.md#server-version) 设置为 `"5.7.25-TiDB-v7.5.0"`。这是因为，当 TiDB 服务器报告版本为 MySQL 8.0.11 或以上时，MySQL Connector/J 会尝试访问 TiDB v7.5.0 不存在的 `information_schema.KEYWORDS` 表。

```

**Comment by Oreoxmt** [comment]
@yusra1007 Thanks for your contribution. I've added a few comments. PTAL

**Review by Oreoxmt — APPROVED** [review]

File: develop/dev-guide-choose-driver-or-orm.md [hunk 5]
```diff
@@ -29,9 +29,10 @@ TiDB 兼容 MySQL 的协议，但存在部分与 MySQL 不兼容或有差异的
 
 按照 [MySQL 文档](https://dev.mysql.com/doc/connector-j/en/)中的说明下载并配置 Java JDBC 驱动程序即可使用。对于 TiDB v6.3.0 及以上版本，建议使用 MySQL Connector/J 8.0.33 及以上版本。
 
-> **建议：**
+> **注意：**
 >
-> 在 8.0.32 之前的 MySQL Connector/J 8.0 版本中存在一个 [bug](https://bugs.mysql.com/bug.php?id=106252)，当与 TiDB v6.3.0 之前的版本一起使用时，可能会导致线程卡死。为了避免此问题，建议使用 MySQL Connector/J 8.0.32 或更高版本，或者使用 TiDB JDBC（见 *TiDB-JDBC* 标签）。
+> - 在 8.0.32 之前的 MySQL Connector/J 8.0 版本中存在一个 [bug](https://bugs.mysql.com/bug.php?id=106252)，当与 TiDB v6.3.0 之前的版本一起使用时，可能会导致线程卡死。为了避免此问题，建议使用 MySQL Connector/J 8.0.32 或更高版本，或者使用 TiDB JDBC（见 *TiDB-JDBC* 标签）。
+> - MySQL Connector/J 8.0 与 TiDB v7.5.0 一起使用时，建议将 TiDB 配置项 [`server-version`](/tidb-configuration-file.md#server-version) 设置为 `"5.7.25-TiDB-v7.5.0"`。这是因为，当 TiDB 服务器报告版本为 MySQL 8.0.11 或以上时，MySQL Connector/J 会尝试访问 TiDB v7.5.0 不存在的 `information_schema.KEYWORDS` 表。
```
Comment (hfxsd) [inline develop/dev-guide-choose-driver-or-orm.md:35]:
```suggestion
> - MySQL Connector/J 8.0 与 TiDB v7.5.0 一起使用时，建议将 TiDB 配置项 [`server-version`](/tidb-configuration-file.md#server-version) 设置为 `"5.7.25-TiDB-v7.5.0"`。这是因为，当 TiDB 服务器报告版本为 MySQL 8.0.11 或更高版本时，MySQL Connector/J 会尝试访问 TiDB v7.5.0 中不存在的 `information_schema.KEYWORDS` 表。
```

**Review by hfxsd — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@yusra1007 Could you please sign the CLA? https://cla-assistant.io/pingcap/docs-cn

**Pull request merged**